tensorflow_encrypted/protocol/pond.py

```diff
@@ -1563,6 +1583,26 @@ def _sum_masked(prot: Pond,
     return prot.sum(x.unmasked, axis, keepdims)
 
 
+#
+# equal helpers
+#
+
+def _equal_public_public(prot, x, y):
+    assert isinstance(x, PondPublicTensor), type(x)
+    assert isinstance(y, PondPublicTensor), type(y)
+
+    x_on_0, x_on_1 = x.unwrapped
+    y_on_0, y_on_1 = y.unwrapped
+
+    with tf.name_scope('equal'):
+        with tf.device(prot.server_0.device_name):
+            z_on_0 = tf.equal(x_on_0.value, y_on_0.value)
+
+        with tf.device(prot.server_1.device_name):
+            z_on_1 = tf.equal(x_on_1.value, y_on_1.value)
+
+    return PondPublicTensor(prot, prot.tensor_factory.Tensor.from_native(z_on_0), prot.tensor_factory.Tensor.from_native(z_on_1), x.is_scaled)
```

Comment:
you have to do the whole `from_native` thing on the two servers, otherwise the underlying operations (eg in case the crt tensor is used) are not pinned to them correctly

Reply:
actually, these should be forwarded to `where` calls on the tensors, which will know how to correctly interpret their values -- the above won't work for crt tensors